nx-X11/programs/Xserver/hw/nxagent/Display.c

```diff
@@ -1907,8 +1918,8 @@ Bool nxagentMakeIcon(Display *display, Pixmap *nxIcon, Pixmap *nxMask)
     agentIconData=nxagentIconData;
   }
 
-
-  snprintf(default_path, PATH_MAX-1, "/usr/NX/share/images/%s", agent_icon_name);
+  /* FIXME: use a compile time define here, /usr/NX is a nomachine path */
+  snprintf(default_path, sizeof(default_path), "/usr/NX/share/images/%s", agent_icon_name);
```

Comment:
I agree, but wonder why the original code used `PATH_MAX - 1`. Probably didn't know that `snprintf` includes the terminating `0` character in the length and only writes `n - 1` elements to the target buffer (+ terminating `0`). `strncpy` is different in this regard. They probably got confused by that...

Reply:
that's what I thought, too.